pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -115,6 +117,69 @@ def test_groupby_aggregation_mixed_dtype():
     tm.assert_frame_equal(result, expected)
 
 
+def test_groupby_aggregation_non_numeric_dtype():
+    # GH #43108
+    df = DataFrame(
+        [["M", [1]], ["M", [1]], ["W", [10]], ["W", [20]]], columns=["MW", "v"]
+    )
+
+    expected = DataFrame(
+        {
+            "v": [[1, 1], [10, 20]],
+        },
+        index=Index(["M", "W"], dtype="object", name="MW"),
+    )
+
+    with tm.assert_produces_warning(UserWarning):
+        g = df.groupby(by=["MW"])
+        result = g.sum()
+        tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_aggregation_multi_non_numeric_dtype():
+    # GH #42395
+    df = DataFrame(
+        {
+            "x": [1, 0, 1, 1, 0],
+            "y": [Timedelta(i, "days") for i in range(1, 6)],
+            "z": [Timedelta(i * 10, "days") for i in range(1, 6)],
+        }
+    )
+
+    expected = DataFrame(
+        {
+            "y": [Timedelta(i, "days") for i in range(7, 9)],
+            "z": [Timedelta(i * 10, "days") for i in range(7, 9)],
+        },
+        index=Int64Index([0, 1], dtype="int64", name="x"),
+    )
+
+    with tm.assert_produces_warning(UserWarning):
+        g = df.groupby(by=["x"])
+        result = g.sum()
+        tm.assert_frame_equal(result, expected)
```

Comment:
Nit: only the `result = g.sum()` should be in the context manager; other lines should be before/after it as this is where the warning is supposed to be generated.